src/main/scala/where/Condition.scala

```diff
@@ -36,13 +36,21 @@ case class NullCondition(reference: Reference, check: NullCheck) extends Conditi
     reference.getReferenceable.toSet
 }
 
-case class PredicateCondition(predicate: Predicate, projection: Collection, where: AnyNode => Condition) extends Condition {
+case class PredicateCondition(
+  predicate: Predicate,
+  projection: Collection,
+  where: ObjectReference => Where
```

Comment:
this gets tricky with the type system... technically these types of conditions can give you a list of nodes or relationships. wrapping in an `ObjectReference` means that the type of the inner thing is just a `Referenceable`, but i still need to pass a concrete anonymous inner thing so i use an `AnyNode` on line 46. any ideas?